C++14/Maths/Basics/FastDoubling_Method.cpp

```diff
@@ -0,0 +1,42 @@
+#include <bits/stdc++.h>
+using namespace std;
+#define ll long long int
+#define MOD 1000000007;
+ll a,b,c,d;
+
+void fast_fib(ll n, ll ans[])
+{
+if(n == 0)
+{
+ans[0] = 0;
+ans[1] = 1;
+return;
+}
+fast_fib((n/2),ans);
+a = ans[0]; /* F(n) */
+b = ans[1]; /* F(n+1) */
+c = 2*b - a;
+if(c < 0)
+c += MOD;
+c = (a * c) % MOD; /* F(2n) */
+d = (a*a + b*b) % MOD; /* F(2n + 1) */

```

Comment:
Mention the formula and logic used and link from where it is taken for readability for other readers.